adminSiteServer/apiRouter.ts

```diff
@@ -664,6 +730,284 @@ apiRouter.get(
+                if (config[key] === "") {
+                    delete config[key]
+                }
+            }
+        })
+
+        await db.transaction(async (t) => {
+            const whereCond1 = suggestedConfigs
+                .map(
+                    (config) =>
+                        `(id = ${escape(
+                            config.id
+                        )} AND config->"$.version" = ${escape(config.version)})`
+                )
+                .join(" OR ")
+            const whereCond2 = suggestedConfigs
+                .map(
+                    (config) =>
+                        `(chartId = ${escape(
+                            config.id
+                        )} AND config->"$.version" = ${escape(config.version)})`
+                )
+                .join(" OR ")
+            // retrieves original chart configs
+            let rows: any[] = await t.query(
+                `
+                SELECT id, config, 1 as priority
+                FROM charts 
+                WHERE ${whereCond1}
+                
+                UNION
+                
+                SELECT chartId as id, config, 2 as priority
+                FROM chart_revisions
+                WHERE ${whereCond2}
+
+                ORDER BY priority
+                `
+            )
+
+            // drops duplicate id-version rows (keeping the row from the
+            // `charts` table when available).
+            rows = rows.filter(
+                (v, i, a) =>
+                    a.findIndex(
+                        (el) =>
+                            el.id === v.id &&
+                            el.config.version === v.config.version
+                    ) === i
+            )
+            if (rows.length < suggestedConfigs.length) {
+                throw new JsonError(
+                    "Failed to retrieve one or more chart ids. Are you certain that all chart id-version combinations are valid?"
+                )
+            } else if (rows.length > suggestedConfigs.length) {
+                throw new JsonError(
+                    "Retrieved more chart configs than expected. This may be due to a bug on the server."
+                )
+            }
+            const originalConfigs: Record<string, GrapherInterface> =
+                rows.reduce(
+                    (obj: any, row: any) => ({
+                        ...obj,
+                        [row.id]: JSON.parse(row.config),
+                    }),
+                    {}
+                )
+
+            // some chart configs do not have an `id` field, so we check for it
+            // and insert the id here as needed. This is important for the
+            // lodash.isEqual condition later on.
+            // for (const [id, config] of Object.entries(originalConfigs)) {
+            //     if (config.id === null || config.id === undefined) {
+            //         config.id = id
+            //     }
+            // }
```

Comment:
Why did you decide to comment this out? It seems to me that this should be safe to do and it would fix the issue with chart 5327 (and the 160 other charts that don't have the id set correctly)

Reply:
fixed in 82c35fa. I had written it originally as a quick fix, then commented it out thinking it would be better to fix the underlying bug before this PR (which didn't happen, and then I forgot this about this commented block).